memory/dma_memtest.py

```diff
@@ -49,7 +49,11 @@ def setUp(self):
         copies of the linux kernel that will be uncompressed.
         """
         self.nfail = 0
-        self.tmpdir = data_dir.get_tmp_dir()
+        base_dir = self.teststmpdir
+        if disk.freespace("/home") > disk.freespace(self.teststmpdir):
```

Comment:
why hard code may this can be yaml input

Reply:
the `dir_to_extract` already is an input, This is more of what can be a default value.

Reply:
yes i got that why we are comparing with /home if as user mounted a device in /mnt or /media so and using tis above logic i can use my mounted parition to create tempfile

Reply:
so i am saying default can be /home but we can provide this as parm as YAML input

Reply:
you want me to interchange `/home` and `self.teststmpdir`?

Reply:
@PraveenPenguin Please let know what is required here.

Reply:
no what i am saying base dir can be user input

Reply:
Is it not an input already here? https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/memory/dma_memtest.py#L53

Reply:
@PraveenPenguin Anything here would be helpful.

Reply:
ah no i got ur point , sorry i missed this disscuion , thanks for explanation